index.js

```diff
@@ -213,25 +212,36 @@ function isFastifyMultipartRegistered (fastify) {
   return fastify.hasContentTypeParser('multipart') && fastify.hasRequestDecorator('multipart')
 }
 
-function createRequestRetry (requestImpl, reply, retriesCount, retryOnError) {
+function createRequestRetry (requestImpl, reply, retriesCount, retryOnError, retryOnCode = 503) {
   function requestRetry (req, cb) {
     let retries = 0
 
     function run () {
       requestImpl(req, function (err, res) {
-        if (err && !reply.sent && retriesCount > retries) {
-          if (err.code === retryOnError) {
-            retries += 1
+        // Magic number, so why not 42? We might want to make this configurable.
+        let retryAfter = 42 * Math.random() * (retries + 1)
 
-            run()
-            return
-          }
+        if (res && res.headers['retry-after']) {
+          retryAfter = res.headers['retry-after']
         }
 
+        if (!reply.sent) {
+          // always retry on 503 errors
+          if (res && res.statusCode === retryOnCode && req.method === 'GET') {
```

Comment:
if a server is returning 503 forever we will be stuck here. Maybe should check the retriesCount here also?

```suggestion
          if ((res && res.statusCode === retryOnCode && req.method === 'GET') || (retriesCount > retries && err && err.code === retryOnError)) {
```

Reply:
Good point! 

I updated the code and made a test for this case... although it doesn't work as expected and does not return, I cannot see why. 
@mcollina @Eomm need another pair of eyes [here](https://github.com/fastify/fastify-reply-from/blob/5628d50a52a05980b277a44f63d45e09c1141260/index.js#L217)

Reply:
It could be a timeout issue
The 10 retries take ~5-10 seconds, so I would reduce the `res.setHeader('Retry-After', 500)` or increase the `tap` timeout

Reply:
I already put a lower timeout with no success

Basically, this plugin will try 10 times (hardwired value) before moving on, if the target server returns 503 for `<= 10` times, it works... if it returns 503 for `> 10` it doesn't

Reply:
The problem is `got` hold the session forever.

Reply:
Fixed!